cpp/cmake/3rdparty/ExternalProjectHelper.cmake

```diff
@@ -0,0 +1,38 @@
+find_package(Git QUIET)
+
+set(THIRD_PARTY_DIRECTORY
+    "${PROJECT_SOURCE_DIR}/3rdparty"
+    CACHE PATH "The path were all the 3rd party projects can be found")
+
+# initialize submodule with given path
+function(initialize_submodule path)
+  if(NOT ${GIT_FOUND})
+    message(
+      FATAL_ERROR "git not found and ${path} sub-module not cloned (use git clone --recursive)")
+  endif()
+  message(STATUS "Sub-module : missing ${path}: running git submodule update --init --recursive")
+  execute_process(
```

Comment:
The exit statut of `execute_process` should be checked and CMake stopped if something bad happens while fetching the git submodule.